src/Nocarrier/HalLink.php

```diff
@@ -80,6 +80,31 @@ public function getAttributes()
     }
 
     /**
+     * Set a particular attribute on this link.
+     *
+     * @param string $name
+     *   Name of the attribute to set.
+     * @param mixed $value
+     *   Value to set the attribute to.
+     * @return void
+     */
+    public function setAttribute($name, $value)
```

Comment:
If we are going down this route, then I think this should have an `overwrite` flag. Mutating state after the fact should require people to be very explicit about what they are doing.